src/main.rs

```diff
@@ -5,93 +5,106 @@ extern crate clap;
-            println!("Unknown config key \"{}\" found for [package.metadata.gh-pages]",
-                     invalid_key);
+fn execute(args: &ArgMatches) -> Result<i32, error::FatalError> {
+    let properties: Table = try!(config::parse_config());
+
+    // Verify the TOML configuration.
+    let valid_keys = vec![config::DOCS_PATH,
+                          config::COMMIT_MESSAGE,
+                          config::SIGN_COMMIT,
+                          config::PUSH_REMOTE,
+                          config::PUSH_BRANCH];
+
+    for key in properties.keys() {
+        if !valid_keys.contains(&key.as_ref()) {
+            println!("Unknown config key \"{}\" found for [package.metadata.sphinx]",
+                     key);
+            return Ok(109);
         }
-        return Ok(109);
     }
 
     // Find parameters or use defaults.
     let dry_run = args.is_present("dry-run");
+
+    let docs_path = args.value_of("docs-path")
+        .or_else(|| config::get_str(&properties, config::DOCS_PATH))
+        .unwrap_or("docs");
+
+    let push = args.is_present("push");
+
+    let commit_msg = args.value_of("commit-message")
+        .or_else(|| config::get_str(&properties, config::COMMIT_MESSAGE))
+        .unwrap_or("(cargo-sphinx) Generate docs.");
     let sign = args.is_present("sign") ||
-               config::get_release_config(&cargo_file, config::SIGN_COMMIT)
-        .and_then(|f| f.as_bool())
-        .unwrap_or(false);
-    let git_remote = args.value_of("push-remote")
-        .or_else(|| {
-            config::get_release_config(&cargo_file, config::PUSH_REMOTE).and_then(|f| f.as_str())
-        })
+               config::get_bool(&properties, config::SIGN_COMMIT).unwrap_or(false);;
+
+    let push_remote = args.value_of("push-remote")
+        .or_else(|| config::get_str(&properties, config::PUSH_REMOTE))
         .unwrap_or("origin");
-    let doc_branch = args.value_of("doc-branch")
-        .or_else(|| {
-            config::get_release_config(&cargo_file, config::DOC_BRANCH).and_then(|f| f.as_str())
-        })
+    let push_branch = args.value_of("push-branch")
+        .or_else(|| config::get_str(&properties, config::PUSH_BRANCH))
         .unwrap_or("gh-pages");
-    let doc_commit_msg = args.value_of("doc-commit-message")
-        .or_else(|| {
-            config::get_release_config(&cargo_file, config::DOC_COMMIT_MESSAGE)
-                .and_then(|f| f.as_str())
-        })
-        .unwrap_or("(cargo-gh-pages) Generate docs.");
-
-    // Check if working directory is clean.
-    if !try!(git::status()) {
-        println!("Uncommitted changes detected, please commit before release");
-        if !dry_run {
-            return Ok(101);
-        }
-    }
 
     // Generate and upload documentation.
-    println!("Building and exporting docs.");
-    try!(cargo::doc(dry_run));
+    println!("Building Sphinx docs.");
+    try!(call(vec!["make", "clean", "html"], docs_path, dry_run));
+
+    if push {
```

Comment:
Bit too long if branch for my taste. What about a separate function just for the pushing?